### DISCUSSION TITLE
[Kyverno] Dynamic approvals for Kubernetes resources

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#600] [Google Summer of Code 2022]
Kyverno is a Kubernetes native policy engine that makes it easy to validate, mutate, and generate resources. While Kyverno policies allow configuratrion of exceptions, excluding policy enforcement on select resources requires manual configuration by cluster administrators. This feature extends Kyverno to allow users to request policy exceptions, and for administrative approvals to be granted asynchronously. The goal of this feature is to allow easier collaboration across developers and operations teams, and to make it easier to manage policies at scale.

Mentor(s): Shuting Zhao (@realshuting), Jim Bugwadia (@JimBugwadia)

https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#dynamic-approvals-for-kubernetes-resources

**Comment by shreemaan-abhishek** [comment]
Hello @realshuting and @JimBugwadia, this issue sounds pretty interesting to me. I did my study on this and so far I have found out that I need to study about:

- the what, why, and how's of Policy Reports 
- validate policies
- understand how CRD's are defined and linked with accompanying schema written in Golang
- How schema changes in Golang are reflected in changes in CRD's

Could you please let me know if I am missing anything? Also, I have asked some questions in the issue conversation, I would be eternally grateful if y'all could answer those.
